library/spdm_requester_lib/libspdm_req_get_version.c

```diff
@@ -126,14 +115,14 @@ return_status libspdm_try_get_version(libspdm_context_t *spdm_context,
                                                   spdm_response.version_number_entry_count);
     if (result == false) {
         libspdm_reset_message_a(spdm_context);
-        return RETURN_DEVICE_ERROR;
-    } else {
-        libspdm_copy_mem(&(spdm_context->connection_info.version),
-                         sizeof(spdm_context->connection_info.version),
-                         &(common_version),
-                         sizeof(spdm_version_number_t));
+        return LIBSPDM_STATUS_NEGOTIATION_FAIL;
     }
 
+    libspdm_copy_mem(&(spdm_context->connection_info.version),
+                     sizeof(spdm_context->connection_info.version),
+                     &(common_version),
+                     sizeof(spdm_version_number_t));
+
     if (version_number_entry_count != NULL && version_number_entry != NULL) {
```

Comment:
@jyao1 I am slightly confused by this code. This is only to be used for testing purposes? Are you assuming that the compiler will remove this code if it's not being used for testing?

Reply:
I will expose the libspdm_get_version() API in requesterlib.

As such, a consumer may want use libspdm_get_version(), libspdm_get_capabilities(), libspdm_negotiate_algorithms() directly instead of libspdm_init_connection()